### ISSUE TITLE
ENA Taxonomy

### ISSUE THREAD

**Issue by nickdos** [ARGA-Genomes/arga-data#11]
Data download: https://ftp.ebi.ac.uk/pub/databases/ena/taxonomy/

DwCA download: https://ftp.ebi.ac.uk/pub/databases/ena/taxonomy/sdwca/

**Comment by nickdos** [comment]
It seems NCBI and ENA share data and use the same taxonomy so might be a duplicate of #9.

**Comment by nickdos** [comment]
I've looked at the IDs for ENA taxonomy and NCBI taxonomy and they use the same taxon IDs. 

ENA provides a DwCA form of the data so I'm attempting to merge that in with the other ALA names sources using the [ALA name matching index code](https://github.com/AtlasOfLivingAustralia/ala-name-matching). 

[GBIF have a utility](https://github.com/gbif/ncbi-dwca) that pulls the NCBI taxonomy from the FTP site and creates a DwCA version. However when I ran this, it produced errors and I could see the data files had issues (extra internal separators that weren't escaped. So I've just gone with the pre-built ENA version for now.

**Comment by nickdos** [comment]
Update. I’ve done a first pass import of the ENA taxonomy (same as NCBI) into the ALA’s [name matching index ](https://github.com/AtlasOfLivingAustralia/ala-name-matching)and I’ve come to the conclusion that its probably not worth doing. The main reason being is the ENA/NCBI taxonomy does not follow the standard taxonomy nomenclatural codes for names - in other words a lot of the names are not in the proper format, particularly for virus and bacterial names. These result in lots of warnings by the ALA names indexing process and don't get matched to anything. 

It probably makes sense to load the NVBI/ENA taxonomy during the pre-ingestion processing, so that we can exclude records based on that taxonomy (e.g.viruses and bacteria, etc.). I think we should be doing a merge on taxonID with a few of the fields (rank, kingdom, phylum, order, class and genus) in the NCBI taxonomy in the code that builds the CSV files. I.e. the same as we’re doing with BioSample and BioProject data. Then during the Pipelines processing, the higher taxa can be passed into the name matching service, which should prevent the spurious matching we've seen for some names.

**Comment by KathrynHallARGA** [comment]
Excellent Nick - I like this approach.  It is good that it allows you the flexibility to make that higher order filtering (vertebrate, bacteria, etc.), but doesn't lock you into something non-Linnaean.  I'd prefer to follow GBIF and ALA as much as we can.  GBIF does bring in NCBI - is there something in the GBIF backbone we can apply - or could be just bring in the GBIF backbone instead?

**Comment by nickdos** [comment]
Yeah, I'll take a look at the GBIF backbone, as it might be the easiest way to allow matching on some of these names.

The fact that the NCBI taxonomy is so messy could be a pro for us, as it might solve the problem you mentioned in a meeting where 2 researchers are handling an un-described taxon - the NCBI taxonomy has a very low barrier of entry and so researchers are more likely to add a new name and then refer to it, than via traditional sources. Most of the virus names appear to be researcher-invented in NCBI already.

**Comment by KathrynHallARGA** [comment]
Absolutely.  GBIF does bring in all the BINs from Barcode of Life and OTUs from NCBI, so it might be the right track.  Also, they have the IUCN lists on there and some other lists too, so it might help circumvent the list issue too.

There is a lot to like on the GBIF occurrence search and display.  It might be good for us to work up the features we like and which we can look to replicate.

**Comment by nickdos** [comment]
KH passed on this project that creates a fully formed tree for the NCBI taxonomy data.

https://github.com/tetsufmbio/taxallnomy

It an produce a `tsv` output file, which looks useful in augmenting sequence data with higher taxa.
